### ISSUE TITLE
consolidation bug 

### ISSUE THREAD

**Issue by spacejam** [spacejam/sled#717]
```
test parallel_tree_ops ... test parallel_tree_ops has been running for over 60 seconds
The application panicked (crashed).
Message:  tried to consolidate set at key <= hi.Set([0, 0, 1, 145], [0, 1, 145]) to node Node { lo: [] hi: [0, 0, 100] next: Some(11) merging_child: None merging: false data: Leaf([([0, 0, 0], [0, 0, 0]), ([0, 0, 1], [0, 0, 1]), ([0, 0, 2], [0, 0, 2]), ([0, 0, 3], [0, 0, 3]), ([0, 0, 4], [0, 0, 4])]) }
Location: crates/sled/src/node.rs:67
```

**Comment by spacejam** [comment]
encountered on #716

**Comment by spacejam** [comment]
this happens because the merge happens without doing a pre-check on the current ts, and it may merge a frag to a totally different page

**Comment by spacejam** [comment]
fixed in #716

**Comment by spacejam** [comment]
fixed in #716
